weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTable.tsx

```diff
@@ -279,6 +279,11 @@ export const CallsTable: FC<{
       setCallsResult([]);
       setCallsTotal(0);
       callsEffectiveFilter.current = effectiveFilter;
+      // Refetch the calls IFF the filter has changed, this is a
+      // noop if the calls query is already loading, but if the filter
+      // has no effective impact (frozen vs. not frozen) we need to
+      // manually refetch
+      calls.refetch();
```

Comment:
When this runs, won't this cause `calls` to change and fire the `useEffect` again?

Reply:
I don't think so, because this useEffect should fire the next render after useCalls is called, which would make this `calls.refetch` a noop, as the calls query will **always** be loading at that point. However, even if this does cause the useEffect to fire again, the `callsEffectiveFilter` ref should now be changed, so either calls is `loading` in which case we do nothing, or it has returned, in which case we set the calls and load the table.